src/components/LeafletMap.js

```diff
@@ -0,0 +1,398 @@
+/* This component will be for the leaflet map and all of its logic
+We anticipate calling NDVIMultiYearGraph from this component
+MapLegend.js will be in here too
+*/
+
+import React, {useDebugValue, useEffect, useRef, useState} from 'react';
+import { NDVIMultiYearGraph } from "./NDVIMultiYearGraph";
+import {Grid} from "@material-ui/core";
+import {MapContainer, useMap} from "react-leaflet";
+import 'leaflet-loading';
+import 'leaflet-loading/src/Control.Loading.css';
+import {geosearch} from "esri-leaflet-geocoder";
+import L from "leaflet";
+import forwarn2Legend from "../forwarn2-legend.png";
+import config from "../config";
+import {getNextFWDate, toWMSDate} from "../datemanagement";
+import {parse} from "fast-xml-parser";
+import { useStateWithLabel, useCompare } from "../utils";
+
+// Map Defaults
+const center = [35, -82]
+const zoom = 13
+
+
+function MapController ({ graphOn, currentGraphCoords, setMap, modisData, setModisData, modisDataConfig,
```

Comment:
We should change this to props and assign them and validate them through the following:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment
https://reactjs.org/docs/typechecking-with-proptypes.html

I can show you more about this but we should do this for all of the components.